### PR TITLE
Overview for docs introduction section

### PR DESCRIPTION
**Pull request by hasheddan**
### Description of your changes

Adds overview for `Introduction` section of docs.

### How has this code been tested?

[skip ci]

### Checklist

I have:
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Ensured this PR contains a neat, self documenting set of commits.
- [x] Updated any relevant [documentation] and [examples].
- [ ] Reported all new error conditions into the log or as an event, as
  appropriate.

For more about what we believe makes a pull request complete, see our
[definition of done].

[documentation]: https://github.com/crossplane/crossplane/tree/master/docs
[examples]: https://github.com/crossplane/crossplane/tree/master/cluster/examples
[definition of done]: https://github.com/crossplane/crossplane/tree/master/design/one-pager-definition-of-done.md

### CODE REVIEW

File: docs/overview.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+life-cycle of those resources. You can read more about providers, including how
+to install and configure them, in the [providers documentation].
+
+## Managed Resources
+
+Managed resources are Kubernetes custom resources that represent infrastructure
+primitives. Managed resources with an API version of `v1beta1` or higher support
+every field that the cloud provider does for the given resource. You can find
+the Managed Resources and their API specifications for each provider on
+[doc.crds.dev].
```
Comment (prasek) [inline docs/overview.md:42]:
```suggestion
[doc.crds.dev] and learn more in the [managed resource documentation](https://github.com/crossplane/crossplane/blob/master/docs/managed-resources.md).
```

File: docs/overview.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+# Overview
+
+Crossplane introduces multiple building blocks that enable you to provision,
+publish, and consume infrastructure using the Kubernetes API. These individual
+concepts work together to allow for powerful separation of concern between
+different personas in an organization, meaning that each member of a team
+interacts with Crossplane at an appropriate level of abstraction.
+
+![Crossplane Concepts]
```
Comment (prasek) [inline docs/overview.md:16]: Might move this **after** the key concepts (i.e. after the OAM section) are introduced and then provide a **text walkthrough of the diagram** building on the concepts introduced on the docs/readme.md
* **Provision & manage cloud infrastructure with kubectl**
  * [Install Crossplane] to provision and manage cloud infrastructure and
    services from any Kubernetes cluster.
  * Provision infrastructure primitives from any provider ([GCP], [AWS],
    [Azure], [Alibaba], on-prem) and use them alongside existing application
    configurations.
  * Version, manage, and deploy with your favorite tools and workflows that
    you’re using with your clusters today.

* **Publish custom infrastructure resources for your applications to use**
  * Define, compose, and publish your own [infrastructure resources] with
    declarative YAML, resulting in your own infrastructure CRDs being added to
    the Kubernetes API for applications to use.
  * Hide infrastructure complexity and include policy guardrails, so
    applications can easily and safely consume the infrastructure they need,
    using any tool that works with the Kubernetes API.
  * Consume infrastructure resources alongside any Kubernetes application to
    provision and manage the cloud services they need with Crossplane as an
    add-on to any Kubernetes cluster.

* **Deploy applications using a team-centric approach with OAM**
  * Define cloud native applications and the infrastructure they require with
    the Open Application Model ([OAM]).
  * Collaborate with a team-centric approach with a strong separation of
    concerns:
    * Infrastructure operators - provide infrastructure and services for
      applications to consume
    * Application developers - build application components independent of
      infrastructure
    * Application operators - compose, deploy, and run application
      configurations
  * Deploy application configurations from app delivery pipelines or GitOps
    workflows, using the proven Kubernetes declarative model.

Comment (prasek) [reply docs/overview.md:16]: So basically a text walkthrough of the diagram showing more concretely, using the example presented the diagram, for how Crossplane enables you to do all these things.

File: docs/overview.md (hunk 1 above)
Comment (hasheddan) [reply docs/overview.md:42]: oh I missed that this got created!

File: docs/overview.md (hunk 2 above)
Comment (hasheddan) [reply docs/overview.md:16]: @prasek that run-down is pretty much verbatim what is on the docs `README.md`, which is the first page that folks are going to see. Are we wanting to reproduce it here as well?

Comment (prasek) [reply docs/overview.md:16]: No not repro, just a walkthrough of the diagram touching on the points covered in the readme, i.e. who does what. Essentially the voiceover for the summary diagram if you were going to explain it to someone.

File: docs/overview.md [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+[managed resources]: #managed-resources
+[OAM]: #oam
+[CRDs]: https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/
+[controllers]: https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/#custom-controllers
+[providers documentation]: providers.md
+[doc.crds.dev]: https://doc.crds.dev
+[managed resources documentation]: managed-resources.md
+[composition documentation]: composition.md
+[Open Application Model]: https://oam.dev/
+[OAM documentation]: oam.md
```
Comment (hasheddan) [inline docs/overview.md:134]: Dependent on @negz naming of this doc

Comment (jbw976) [reply docs/overview.md:134]: is this needed before merge @hasheddan @prasek?

Comment (hasheddan) [reply docs/overview.md:134]: I think we can go ahead and update if it doesn’t match 👍

**Pull request merged**